specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,977 @@
+
+## Mandatory Capabilities
+
+### Bullet 1: Quantum transformations
+
+The set of available instructions that transform the quantum state may vary
+depending on the targeted backend. The profile specification defines how to
+leverage and combine the available instructions to express a program, but does
+not dictate which quantum instructions may be used. Targeting a program to a
+specific backend requires choosing a suitable profile and quantum instruction
+set (QIS). Both can be chosen largely independently, though certain instruction
+sets may be incompatible with this (or other) profile(s). The section on the
+[quantum instruction set](#quantum-instruction-set) defines the requirements for
+a QIS to be compatible with the adaptive profile.
+More information about the role of
+the QIS, recommendations for front- and backend providers, as well as the
+distinction between run-time functions and quantum instructions can be found in this
+[document](../Instruction_Set.md).
+
+### Bullet 2: Measurements
+
+The second requirement relieves a restriction from the base profile that qubits
+can only be measured at the end of the program and that no quantum operations
+can be performed on a measured qubit.
+The only requirement here is that all qubits must be measurable, but
+there are no restrictions on when these measurements are made, and there is
+no restriction necessitating that all qubits be measured. I.E. it's perfectly
+valid to measure a subset of qubits used in the program, not have all measured
+qubits in the output, and perform the measurements at different points.
+
+### Bullet 3: Program output
+
+There is no change to output specifications with respect to the base profile.
+
+The specification of QIR and all its profiles needs to accurately
+reflect the program intent. This includes being able to define and customize the
+program output. The base profile  and adaptive profile specifications hence require explicitly
+expressing which values/measurements are returned by the program and in which
+order. How to express this is defined in the section on [output
+recording](#output-recording).
+
+A suitable output schema can be
+generated in a post-processing step after the computation on the quantum
+processor itself has completed; customization of the program output hence does
+not require support on the QPU itself.
+
+The defined [output schemas](../output_schemas/) provide different options for
+how a backend may express the computed value(s). The exact schema can be freely
+chosen by the backend and is identified by a string label in the produced
+schema. Each output schema contains sufficient information to allow quantum
+programming frameworks to generate a user-friendly presentation of the returned
+values in the requested order, such as, e.g., a histogram of all results when
+running the program multiple times.
+
+### Bullet 4: Conditional Forward branching
```

Comment:
nit: Similar to another comment, since branching already implies a condition, maybe this can be just "Forward Branching".